docs/design.md

```diff
@@ -1,37 +1,145 @@
+
+When the Service loads this config the result will look like this (part of processors and exporters are omitted from the diagram for brevity):
+
+![Receivers](images/design-receivers.png)
+
+Important: when the same receiver is referenced in more than one pipeline the Service will create only one receiver instance at runtime that will send the data to fanoutprocessor which in turn will send the data to the first processor of each pipeline. The data propagation from receiver to fanoutprocessor and then to processors is via synchronous function call. This means that if one processor blocks the call the other pipelines that are attached to this receiver will be blocked from receiving the same data and the receiver itself will stop processing and forwarding newly received data.
+
+### Exporters
+
+Exporters typically forward the data they get to a destination on a network (but they can also send it elsewhere, e.g “logging” exporter writes the telemetry data to a local file). 
+
+The configuration allows to have multiple exporters of the same type, even in the same pipeline. For example one can have 2 “jaeger” exporters defined each one sending to a different jaeger endpoint, e.g.:
+
+```yaml
+exporters:
+  jaeger/1:
+    protocols:
+      grpc:
+        endpoint: "example.com:14250”
+  jaeger/2:
+    protocols:
+      grpc:
+        endpoint: "127.0.0.1:14250”
+```
+
+Usually an exporter gets the data from one pipeline, however it is possible to configure multiple pipelines to send data to the same exporter, e.g.:
+
+```yaml
+exporters:
+  jaeger:
+    protocols:
+      grpc:
+        endpoint: "127.0.0.1:14250”
+
+pipelines:
+  traces:  # a pipeline of “traces” type
+    receivers: [zipkin]
+    processors: [tags, tail-sampling, batch, queued-retry]
+    exporters: [jaeger]
+  traces/2:  # another pipeline of “traces” type
+    receivers: [opencensus]
+    processors: [batch]
+    exporters: [jaeger]
+```
+
+In the above example “jaeger” exporter will get data from pipeline “traces” and from pipeline “traces/2”. When the Service loads this config the result will look like this (part of processors and receivers are omitted from the diagram for brevity):
+
+![Exporters](images/design-exporters.png)
+
+### Processors
+
+A pipeline can contain one or more sequentially connected processors. The first processor gets the data from one or more receivers that are configured for the pipeline, the last processor sends the data to one or more exporters that are configured for the pipeline. All processors between the first and last receive the data strictly only from one preceding processor and send data strictly only to the succeeding processor.
```

Comment:
May be worth mentioning processors are not necessary (at least in the metrics pipeline).

Reply:
Good point, I will add.

Reply:
Added.